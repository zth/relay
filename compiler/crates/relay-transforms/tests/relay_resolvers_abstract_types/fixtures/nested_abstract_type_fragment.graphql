# relay-resolver-enable-interface-output-type

fragment nestedAbstractTypeFragment on Cat {
    ... on Tabby {
        parent {
            description
        }
    }
}

# %extensions%

interface Cat {
    description: String
}

type Tabby implements Cat {
    description: String
    parent: Cat
}

type Persian implements Cat {
    description: String
}

extend type Query {
    cat: Cat
}
