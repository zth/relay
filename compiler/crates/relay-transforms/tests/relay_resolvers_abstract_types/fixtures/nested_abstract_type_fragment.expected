==================================== INPUT ====================================
# relay-resolver-enable-interface-output-type

fragment nestedAbstractTypeFragment on Cat {
    ... on Tabby {
        parent {
            description
        }
    }
}

# %extensions%

interface Cat {
    description: String
}

type Tabby implements Cat {
    description: String
    parent: Cat
}

type Persian implements Cat {
    description: String
}

extend type Query {
    cat: Cat
}
==================================== OUTPUT ===================================
fragment nestedAbstractTypeFragment on Cat {
  ... on Tabby {
    parent {
      ... on Tabby {
        description
      }
      ... on Persian {
        description
      }
    }
  }
}
